### PR TITLE
fix(common): focus fix on play filter modal open

### LINKED ISSUE #1414: 🐛 [Bug report]: Focus doesn't come on first element of the modal when Filter plays modal is opened
### Descrcibe the bug?

Focus doesn't come on first element of the modal when Filter plays modal is opened

### Steps to reproduce the bug?

1. Go to https://reactplay.io/plays
2. Press tab key 3 times to reach the filter plays icon. Click on it to open the filter plays modal.
3. Notice that the focus stays on the icon and doesn't move to the first input element in the modal.

### Expected behavior

The focus should move to the first element in the modal

### Desktop (Please provide your system information)

Should come on all browsers/OS

### Mobile (Please provide your device information)

_No response_

### Screenshot / Screenshare

_No response_

### Relevant log output

_No response_

### Record

- [X] I have checked the existing [issues](https://github.com/reactplay/react-play/issues)
- [X] I have read the [Contributing Guidelines](https://github.com/reactplay/react-play/blob/main/CONTRIBUTING.md)
- [X] I want to work on this issue

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by hjaintech**
> First thing, PLEASE READ THIS: [ReactPlay Code Review Checklist](https://github.com/reactplay/react-play/wiki/ReactPlay-Code-Review-Checklist)

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #1414 (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Yes. When you open the filter plays modal now, the focus will be on the first element inside the modal by default. Also, when you close the modal, the focus would go back to the button which opens this modal.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

# Screenshots or example output

### CODE REVIEW

File: src/common/modal/index.jsx [hunk 1]
```diff
@@ -1,8 +1,23 @@
...
 
 const Modal = ({ title, show, onClose, onSubmit, children, cname }) => {
+  const modalChildrenRef = useRef(null);
+
+  // useEffect to set the focus on the first focussable element in the modal
+  // when it opens up
+  useEffect(() => {
+    if (show) {
+      const modalChildElements = modalChildrenRef.current;
+      const focusableElements = modalChildElements.querySelectorAll(
```
Comment (priyankarpal) [inline src/common/modal/index.jsx:13]:
```suggestion
      const focusableElements = modalChildElements?.querySelectorAll(
```

File: src/common/modal/index.jsx [hunk 2]
```diff
@@ -1,8 +1,23 @@
...
+        'button, [href], input, select, textarea, [tabindex]:not([tabindex="-1"])'
+      );
+      const firstFocusableElement = focusableElements[0];
+      firstFocusableElement.focus();
```
Comment (priyankarpal) [inline src/common/modal/index.jsx:17]:
``` suggestion
      focusableElements?.[0]?.focus();
```

**Review by priyankarpal — APPROVED** [review]
Looks Good 🎉

**Comment by hjaintech** [comment]
> Looks Good 🎉

@priyankarpal Fixed both of your comments. Can you approve again please?

**Review by priyankarpal — APPROVED** [review]

**Comment by priyankarpal** [comment]
> > Looks Good 🎉
> 
> @priyankarpal Fixed both of your comments. Can you approve again please?

done ✅ Thank you

**Review by atapas — APPROVED** [review]

**Pull request merged**